lib/cast/double.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const assert = require('assert');
+const BSON = require('bson');
+
+/**
+ * Given a value, cast it to a IEEE 754-2008 floating point, or throw an `Error` if the value
+ * cannot be casted. `null`, `undefined`, and `NaN` are considered valid inputs.
+ *
+ * @param {Any} value
+ * @return {Number}
+ * @throws {Error} if `value` does not represent a IEEE 754-2008 floating point. If casting from a string, see [BSON Double.fromString API documentation](https://mongodb.github.io/node-mongodb-native/Next/classes/BSON.Double.html#fromString)
+ * @api private
+ */
+
+module.exports = function castDouble(val) {
```

Comment:
I think this function needs to return a `Double`.  Otherwise, this doesn't work:

```
        await Test.create({ myDouble: '42' });
        const doc = await Test.findOne({ myDouble: { $type: 'double' } });
        assert.deepEqual(doc.myDouble, new BSON.Double(42));
```

because our bSon library inserts myDouble as an int32.

Reply:
(we'll also need tests for this)

Reply:
(( you'll also note that if you run this as a test, it fails - we need to update mongoose's `clone` method too to support Doubles ))

Reply:
added in the clone functionality!